File: code-samples/php/transaction-status.php
<?php
/**
 * KwikAPI SDK — Transaction Status
 * ─────────────────────────────────────────────────────────────────────────────
 * Returns the real-time status of a previously initiated transaction identified by your unique order_id. Always call this after any payment API to confirm the final status (SUCCESS / FAILURE / PENDING). Do not rely solely on the payment API response — network issues can interrupt delivery. Implement a polling or webhook strategy.
 *
 * Endpoint  : GET /api/v2/status.php
 * Group     : Fetching APIs
 * Rate Limit: 100 / day
 * NOTE: Always verify status — never rely solely on payment API response.
 *
 * Parameters:
 *   @param  string   $api_key          (required)  Your KwikAPI API key
 *   @param  string   $order_id         (required)  Your unique order ID used during the payment request
 *
 * @return array  Decoded JSON response from KwikAPI
 * @throws RuntimeException  on cURL failure or invalid JSON response
 *
 * Environment:
 *   UAT (testing) : https://uat.kwikapi.com
 *   Production    : https://www.kwikapi.com  (change KWIKAPI_BASE_URL)
 *
 * @link https://kwikapi.com/auth/register  Get your API key
 */

define('KWIKAPI_BASE_URL', 'https://uat.kwikapi.com'); // Switch to https://www.kwikapi.com for production
define('KWIKAPI_KEY', 'YOUR_API_KEY');

/**
 * Transaction Status
 *
 *   @param  string   $api_key          (required)  Your KwikAPI API key
 *   @param  string   $order_id         (required)  Your unique order ID used during the payment request
 * @return array
 * @throws RuntimeException
 */
function kwik_transaction_status(string $apiKey = KWIKAPI_KEY, string $order_id = 'YOUR_ORDER_ID'): array
{
    $queryString = http_build_query(['api_key' => $apiKey, 'order_id' => $order_id]);
    $ch = curl_init();
    curl_setopt_array($ch, [
        CURLOPT_URL            => 'https://uat.kwikapi.com/api/v2/status.php?' . $queryString,
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_HTTPGET        => true,
        CURLOPT_SSL_VERIFYPEER => true,
        CURLOPT_TIMEOUT        => 30,
        CURLOPT_HTTPHEADER     => ['Accept: application/json'],
    ]);

    $response = curl_exec($ch);
    $httpCode  = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $curlError = curl_error($ch);
    curl_close($ch);

    if ($response === false) {
        throw new RuntimeException("cURL error: {$curlError}");
    }

    $data = json_decode($response, true);
    if (json_last_error() !== JSON_ERROR_NONE) {
        throw new RuntimeException('Invalid JSON response (HTTP ' . $httpCode . ')');
    }

    return $data;
}

// ── Example usage ─────────────────────────────────────────────────────────────
try {
    $result = kwik_transaction_status(KWIKAPI_KEY, 'YOUR_ORDER_ID');

    if ($result['success'] ?? false) {
        echo "Success: " . ($result['message'] ?? 'OK') . PHP_EOL;
        echo json_encode($result, JSON_PRETTY_PRINT) . PHP_EOL;
    } else {
        echo "API Error: " . ($result['message'] ?? 'Unknown error') . PHP_EOL;
    }
} catch (RuntimeException $e) {
    echo "Request failed: " . $e->getMessage() . PHP_EOL;
    exit(1);
}

File: code-samples/python/transaction-status.py
#!/usr/bin/env python3
"""
KwikAPI SDK — Transaction Status
─────────────────────────────────────────────────────────────────────────────
Returns the real-time status of a previously initiated transaction identified by your unique order_id. Always call this after any payment API to confirm the final status (SUCCESS / FAILURE / PENDING). Do not rely solely on the payment API response — network issues can interrupt delivery. Implement a polling or webhook strategy.

Endpoint  : GET /api/v2/status.php
Group     : Fetching APIs
Rate Limit: 100 / day
    # NOTE: Always verify status — never rely solely on payment API response.

Requirements:
    pip install requests

Environment:
    UAT (testing) : https://uat.kwikapi.com
    Production    : https://www.kwikapi.com  (change BASE_URL)

Get your API key: https://kwikapi.com/auth/register
"""

import requests
from typing import Any

BASE_URL = "https://uat.kwikapi.com"  # Switch to https://www.kwikapi.com for production
API_KEY  = "YOUR_API_KEY"


class KwikAPIError(Exception):
    """Raised when the KwikAPI returns a non-success response."""


def transaction_status(api_key: str = 'YOUR_API_KEY', order_id: str = 'YOUR_ORDER_ID') -> dict[str, Any]:
    """
    Transaction Status

    Args:
        api_key (string): Required. Your KwikAPI API key
        order_id (string): Required. Your unique order ID used during the payment request

    Returns:
        dict: Parsed JSON response from KwikAPI.

    Raises:
        KwikAPIError: If the API response wrapper is missing.
        requests.HTTPError: On non-2xx HTTP status.
    """
    url = BASE_URL + "/api/v2/status.php"

    with requests.Session() as session:
        session.headers.update({"Accept": "application/json"})
        response = session.get(url, params={'api_key': api_key, 'order_id': order_id})

    response.raise_for_status()
    data = response.json()

    if "response" not in data:
        raise KwikAPIError(data.get("message", "Unknown API error"))

    return data


# ── Example usage ─────────────────────────────────────────────────────────────
if __name__ == "__main__":
    try:
        result = transaction_status('YOUR_API_KEY', 'YOUR_ORDER_ID')
        import json
        print(json.dumps(result, indent=2))
    except KwikAPIError as e:
        print(f"API Error: {e}")
        raise SystemExit(1)
    except requests.RequestException as e:
        print(f"Request failed: {e}")
        raise SystemExit(1)
